### ISSUE TITLE
sub-TLVs parsing

### ISSUE THREAD

**Issue by boutier** [fingon/pybabel#1]
Sub-TLVs are not well ignored.

For example, I've got:

```
invalid prefix in update request Update(l=29, seqno=41025, flags=192, interval=1600, metric=256, body=b' \x01\x05\xc0\x15\x05k\x00\x02\x1b\xb1\xff\xfe\x83\x8d2\x02\x01$', plen=128, omitted=0, ae=2) (Update(l=29, seqno=41025, flags=192, interval=1600, metric=256, body=b' \x01\x05\xc0\x15\x05k\x00\x02\x1b\xb1\xff\xfe\x83\x8d2\x02\x01$', plen=128, omitted=0, ae=2))
```

on this Update TLV:

```
0000   08 1d 02 c0 80 00 06 40 a0 41 01 00 20 01 05 c0
0010   15 05 6b 00 02 1b b1 ff fe 83 8d 32 02 01 24
```

Were "02 01 24" holds for "channels 1 and 36"...

**Comment by fingon** [comment]
I cannot find definition of sub TLVs in RFC6126.. It says Update's only variable length content is the prefix field ;)

**Comment by jech** [comment]
Section 4.3 says:

```
Length    The length of the body, exclusive of the Type and Length
          fields.  If the body is longer than the expected length of
          a given type of TLV, any extra data MUST be silently
          ignored.
```

The fact that the body is followed by a bunch of sub-TLVs is irrelevant to you — all you need to do is make sure that you ignore any garbage that may follow the body but be within Length octets from the beginning.

You don't need to know about sub-TLVs, but they are defined in https://tools.ietf.org/html/draft-chroboczek-babel-extension-mechanism

**Comment by fingon** [comment]
Oh, right. I will fix it in a bit ;) (Writing MUST/SHOULD unit tests right now just out of perverse interest to see how badly I have done the first iteration.)

**Comment by fingon** [comment]
Fixed in 760f12318d25f96e60201ce5b57d45914c79f850.
